opentelemetry-api/src/opentelemetry/context/context.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import typing
+from abc import ABC, abstractmethod
+
+
+class Context(typing.Dict[str, object]):
+    def __setitem__(self, key: str, value: object) -> None:
+        raise ValueError
+
```

Comment:
This makes a `copy` return a `Context`, making it unnecessary to create a `Context` when returning in `remove_value` and `set_value`.

```suggestion
    def copy(self, **kwargs):
        contents = {key: value for key, value in self.items()}
        contents.update(kwargs)
        return Context(contents)
```